src/lib/provable/gadgets/twisted-curve.ts

```diff
@@ -0,0 +1,513 @@
+import { inverse, mod } from '../../../bindings/crypto/finite-field.js';
+import { Provable } from '../provable.js';
+import { assert } from './common.js';
+import { Field3, ForeignField, split } from './foreign-field.js';
+import { l2Mask } from './range-check.js';
+import { sha256 } from 'js-sha256';
+import { provable } from '../types/provable-derivers.js';
+import {
+  bigIntToBytes,
+  bytesToBigInt,
+} from '../../../bindings/crypto/bigint-helpers.js';
+import {
+  CurveTwisted,
+  GroupTwisted,
+  twistedAdd,
+  twistedDouble,
+} from '../../../bindings/crypto/elliptic-curve.js';
+import { assertPositiveInteger } from '../../../bindings/crypto/non-negative.js';
+import { sliceField3 } from './bit-slices.js';
+import { exists } from '../core/exists.js';
+import { arrayGetGeneric } from './elliptic-curve.js';
+
+// external API
+export { CurveTwisted };
+
+// internal API
+export { Point, initialAggregator, simpleMapToCurve, arrayGetGeneric };
+
+const CurveTwisted = {
+  add,
+  double,
+  negate,
+  assertOnCurve,
+  scale,
+  multiScalarMul,
+};
+
+/**
+ * Non-zero twisted elliptic curve point.
+ */
+type Point = { x: Field3; y: Field3 };
+type point = { x: bigint; y: bigint };
+
+const Point = {
+  from({ x, y }: point): Point {
+    return { x: Field3.from(x), y: Field3.from(y) };
+  },
+  toBigint({ x, y }: Point) {
+    let x_ = Field3.toBigint(x);
+    let y_ = Field3.toBigint(y);
+    return { x: x_, y: y_, infinity: x_ === 0n && y_ === 1n };
```

Comment:
I created an equivalent `Point` type (elliptic-curve also has it) with the difference that the infinity is not always set to `false` as it is the case in the other gadgets.